deployments/rds-s3/terraform/rds-s3-components/main.tf

```diff
@@ -0,0 +1,329 @@
+
+  kfp_chart_map = {
+    (local.kfp_chart_vanilla) = !var.use_rds && !var.use_s3,
+    (local.kfp_chart_rds_only) = var.use_rds && !var.use_s3,
+    (local.kfp_chart_s3_only) = !var.use_rds && var.use_s3,
+    (local.kfp_chart_rds_and_s3) = var.use_rds && var.use_s3
+  }
+
+  katib_chart = var.use_rds ? local.katib_chart_rds : local.katib_chart_vanilla
+  kfp_chart = [for k,v in local.kfp_chart_map : k if v == true][0]
+}
+
+resource "kubernetes_namespace" "kubeflow" {
+  metadata {
+    labels = {
+      control-plane = "kubeflow"
+      istio-injection = "enabled"
+    }
+
+    name = "kubeflow"
+  }
+}
+
+module "kubeflow_secrets_manager_irsa" {
+  source            = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/irsa?ref=v4.9.0"
+  kubernetes_namespace = "kubeflow"
+  create_kubernetes_namespace = false
+  create_kubernetes_service_account = true
+  kubernetes_service_account        = "kubeflow-secrets-manager-sa"
+  irsa_iam_policies                 = ["arn:aws:iam::aws:policy/AmazonSSMReadOnlyAccess", "arn:aws:iam::aws:policy/SecretsManagerReadWrite"]
+  irsa_iam_role_path                = var.addon_context.irsa_iam_role_path
+  irsa_iam_permissions_boundary     = var.addon_context.irsa_iam_permissions_boundary
+  eks_cluster_id                    = var.addon_context.eks_cluster_id
+  eks_oidc_provider_arn             = var.addon_context.eks_oidc_provider_arn
+}
+
+resource "random_password" "db_password" {
+  count = var.generate_db_password ? 1 : 0
+  length           = 16
+  special          = true
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+
+module "rds" {
+  count = var.use_rds ? 1 : 0
+  source            = "../../../../iaac/terraform/aws-infra/rds"
+  vpc_id     = var.vpc_id
+  subnet_ids = var.subnet_ids
+  security_group_id = var.security_group_id
+  db_name = var.db_name
+  db_username = var.db_username
+  db_password = coalesce(var.db_password, random_password.db_password[0].result)
+  db_class = var.db_class
+  db_allocated_storage = var.db_allocated_storage
+  multi_az = var.multi_az
+  secret_recovery_window_in_days = var.secret_recovery_window_in_days
+}
+
+module "s3" {
+  count = var.use_s3 ? 1 : 0
+  source            = "../../../../iaac/terraform/aws-infra/s3"
+  force_destroy_bucket = var.force_destroy_s3_bucket
+  aws_access_key = var.aws_access_key
+  aws_secret_key = var.aws_secret_key
+  secret_recovery_window_in_days = var.secret_recovery_window_in_days
+}
+
+module "filter_secrets_manager_set_values" {
+  source            = "../../../../iaac/terraform/utils/set-values-filter"
+  set_values = {
+    "rds.secretName" = try(module.rds[0].rds_secret_name, null),
+    "s3.secretName" = try(module.s3[0].s3_secret_name, null),
+  }
+}
+
+module "secrets_manager" {
+  count = var.use_rds || var.use_s3 ? 1 : 0
+  source            = "../../../../iaac/terraform/common/aws-secrets-manager"
+  helm_config = {
+    chart = "${var.kf_helm_repo_path}/charts/common/aws-secrets-manager/rds-s3"
```

Comment:
this also needs to change according to deployment option?
https://github.com/awslabs/kubeflow-manifests/tree/main/charts/common/aws-secrets-manager changes according to option

Reply:
fixed